File: backend/python/DumboControllerReset.py
#!/usr/bin/python
from IRobot import IRobot
from random import choice

class DumboControllerReset(object):
    def controlRobot(self,robot):
        # Select a random direction
        #direction = choice([IRobot.AHEAD, IRobot.RIGHT, IRobot.BEHIND, IRobot.LEFT])
          # Face the robot in this direction

        if robot.getHeading()==IRobot.NORTH:
        	robot.face(IRobot.BEHIND)
        elif robot.getHeading()==IRobot.EAST:
        	robot.face(IRobot.RIGHT)
        elif robot.getHeading()==IRobot.SOUTH:
        	robot.face(IRobot.AHEAD)
        elif robot.getHeading()==IRobot.WEST:
        	robot.face(IRobot.LEFT)

        robot.advance()        # and move the robot

    def reset(self):
        print("Resetting the controller")

File: backend/java/controllers/DumboControllerReset.java
public class DumboControllerReset
{
    
    public void controlRobot(IRobot robot) {

        int randno;
        int direction;

        // Select a random number

        randno = (int) Math.round(Math.random()*3);

        // Convert this to a direction

        if (randno == 0)
            direction = IRobot.LEFT;
        else if (randno == 1)
            direction = IRobot.RIGHT;
        else if (randno == 2)
            direction = IRobot.BEHIND;
        else 
            direction = IRobot.AHEAD;
         
        robot.face(direction);  /* Face the robot in this direction */   

        robot.advance();        /* and move the robot */
    }

    public void reset() {
        System.out.println("Resetting the controller");
    }

}